doc/source/data/key-concepts.rst

```diff
@@ -36,75 +37,72 @@ This flexibility is a unique characteristic of Ray Datasets.
 ..
   https://docs.google.com/drawings/d/132jhE3KXZsf29ho1yUdPrCHB9uheHBWHJhDQMXqIVPA/edit
 
-Datasets shuffle can scale to processing hundreds of terabytes of data. See the :ref:`Performance Tips Guide <shuffle_performance_tips>` for an in-depth guide on shuffle performance.
+Datasets can shuffle hundreds of terabytes of data. For an in-depth guide on shuffle performance, read :ref:`Performance Tips and Tuning <shuffle_performance_tips>`.
 
 Execution mode
 ==============
 
-Most Datasets operations are lazily executed. Operations won't be executed until the dataset is consumed or :meth:`ds.fully_executed() <ray.data.Dataset.fully_executed>` is called to manually trigger execution. See the :ref:`Execution Section <datasets_execution>` for more details.
+Most transformations are lazy. They don't execute until you consume a dataset or call
+:meth:`Dataset.fully_executed() <ray.data.Dataset.fully_executed>`.
+
+For an in-depth guide on Datasets execution, read :ref:`Execution <datasets_execution>`.
 
 Fault tolerance
 ===============
 
-Datasets relies on :ref:`task-based fault tolerance <task-fault-tolerance>` in Ray core. Specifically, a Dataset will be automatically recovered by Ray in case of failures. This works through *lineage reconstruction*: a Dataset is a collection of Ray objects stored in shared memory, and if any of these objects are lost, then Ray will recreate them by re-executing the tasks that created them.
+Datasets performs *lineage reconstruction* to recover data. If an application error or
+hardware failure occurs, Datasets recreates lost blocks by re-executing tasks.
 
-There are a few cases that are not currently supported:
+Fault tolerance isn't supported in two cases:
 
 * If the original worker process that created the Dataset dies. This is because the creator stores the metadata for the :ref:`objects <object-fault-tolerance>` that comprise the Dataset.
-* When ``compute=ActorPoolStrategy()`` is specified for transformations.
-
-.. _dataset_pipeline_concept:
-
------------------
-Dataset Pipelines
------------------
-
-Dataset pipelines allow Dataset transformations to be executed incrementally on *windows* of the base data, instead of on all of the data at once. This can be used for streaming data loading into ML training, or to execute batch transformations on large datasets without needing to load the entire dataset into cluster memory.
-
-..
-  https://docs.google.com/drawings/d/1A_nWvignkdvs4GPRShCNYcnb1T--iQoSEeS4uWRVQ4k/edit
-
-.. image:: images/dataset-pipeline-2-mini.svg
-
-Dataset pipelines can be read in a streaming fashion by one consumer, or split into multiple sub-pipelines and read in parallel by multiple consumers for distributed training. See the :ref:`Dataset Pipelines Guide <pipelining_datasets>` for an in-depth guide on pipelining compute.
+* If ``compute=ActorPoolStrategy()`` is specified for transformations. This is because Datasets relies on :ref:`task-based fault tolerance <task-fault-tolerance>`.
```

Comment:
```suggestion
* If  you specify ``compute=ActorPoolStrategy()`` for transformations. This is because Datasets relies on :ref:`task-based fault tolerance <task-fault-tolerance>`.
```

Reply:
Not sure if this mangles the meaning...